### PR TITLE
Automated synchronizing SRPM package via API/CLI/UI

### LINKED ISSUE #3854: Automate synchronizing SRPM package via API/CLI/UI
# Request

Since **pulp** allows one to synchronize `source rpms` (aka `srpms`), we need to make sure that we can perform the synchronization of repositories containing this type of content.

## Setup

* Create a new organization **ORG**
* Create a new lifecycle environment **LCE** with **LIBRARY** as its prior environment
* Create a new product **PROD**
* Create a new cusotm repository **REPO** with a feed pointed at ~~`https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_srpm_repo/`~~  `https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/srpm/` (note the `srpm` part of the url)

## Tests

* Synchronize content and assert that it is fully synchronized and that `srpms` can be listed
* Synchronize content, add product to a content view, publish it and make sure that `srpms` can be listed under the proper content view
* Synchronize content, add product to a content view, publish it, promote it to  and make sure that `srpms` can be listed under the proper content view

## Validation

Since `srpms` are **not** displayed on Satellite interface, there are three possible ways to validate that this type of content has been successfully synchronized:

* Look under `/var/lib/pulp/published/yum/https/repos/ORG/Library/custom/PROD/REPO/` and make sure that you see files with names ending in `.src.rpm`
* Open a connection (GET) against `http://SATELLITE_FQDN/pulp/repos/ORG/Library/custom/PROD/REPO/drpms/` and check that the listing returned contains packages with names ending in `.src.rpm`
* Create a new host and associate it with a product containing `srpm` packages. Run `yum install` or `dnf install` and check that you can install a `srpms`

I recommend we use either the first or second method.

### PR DESCRIPTION
**Pull request by abalakh**
Closes #3854

Test results:
```python
py.test tests/foreman/{api,cli,ui}/test_repository.py -k 'test_sync' -v -n 2
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
[gw0] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw1] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw0] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
[gw1] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
gw0 [9] / gw1 [9]
scheduling tests via LoadScheduling

tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync 
tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_promote_cv 
[gw0] PASSED tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync 
tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_cv 
[gw1] PASSED tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_promote_cv 
tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync 
[gw0] PASSED tests/foreman/api/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_cv 
tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_cv 
[gw1] PASSED tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync 
tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_promote_cv 
[gw0] PASSED tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_cv 
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync 
[gw1] PASSED tests/foreman/cli/test_repository.py::SRPMRepositoryTestCase::test_sync_publish_promote_cv 
[gw0] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync 
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync_publish_promote_cv 
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync_publish_cv 
[gw0] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync_publish_promote_cv 
[gw1] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_sync_publish_cv 

============================================ 9 passed in 517.84 seconds =============================================
```

### CODE REVIEW

**Comment by omaciel** [comment]
@abalakh FANTASTIC!!! The only nitpick I have is related to the names of tests... How would you feel if I asked you to add `positive` to the test names?

**Comment by abalakh** [comment]
@omaciel 😄  now i get why the names looked suspiciously short for me. Thank you!

**Comment by abalakh** [comment]
@ names updated, thanks for bringing it up.

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -1075,3 +1080,106 @@ def test_positive_sync_rh_atomic(self):
...
+        ).create()
+        repo.sync()
+        result = ssh.command(
+            'ls /var/lib/pulp/published/yum/https/repos/{}/Library'
+            '/custom/{}/{}/ | grep .src.rpm'
+            .format(
+                self.org.label,
+                self.product.label,
+                repo.label,
+            )
```
Comment (sghai) [inline tests/foreman/api/test_repository.py]: :+1:

**Comment by sghai** [comment]
ACK..

**Pull request merged**